### ISSUE TITLE
No Key parameter in constructor

### ISSUE THREAD

**Issue by NiklasLehnfeld** [thangmam/smoothratingbar#36]
It is best practice to integrate the Key parameter into all widgets, since it depends on the use case if you need it or not. 
Read more about it [here](https://medium.com/flutter/keys-what-are-they-good-for-13cb51742e7d).

Unfortunately this parameter is missing in the SmoothStarRating class.

**Comment by 6h4n3m** [comment]
I second this, I ran into an issue today while using a SmoothStarRating widget inside a BlocBuilder, I'm querying an API in my bloc and that causes a rebuild based on whether a user's rating changes or not... the SmoothStarRating is the only widget that doesn't rebuild and still uses the old cached rating.. I eventually had to add a Wrapper widget with a unique key in order to solve the issue.

I reckon this happens not just with BloCs but with anyone using StreamBuilders or Provider as well.

**Comment by NiklasLehnfeld** [comment]
Yes, I had the same issue using a StreamBuilder. This is why I made the fork and the PR. Feel free to include my fork as long as this PR is not merged. Using: 
```
smooth_star_rating:
    git:  
      url: https://github.com/NiklasLehnfeld/smoothratingbar
```

**Comment by 6h4n3m** [comment]
> Yes, I had the same issue using a StreamBuilder. This is why I made the fork and the PR. Feel free to include my fork as long as this PR is not merged. Using:
> 
> ```
> smooth_star_rating:
>     git:  
>       url: https://github.com/NiklasLehnfeld/smoothratingbar
> ```

Nice work, hopefully your PR gets merged soon.
